packages/components/src/internal/AssayDefinitionModel.ts

```diff
@@ -261,29 +261,22 @@ export class AssayDefinitionModel extends Record({
         return List(sampleCols.map(this.sampleColumnFieldKey));
     }
 
-    createSampleFilter(
-        sampleColumns: List<string>,
-        value,
-        singleFilter: Filter.IFilterType,
-        whereClausePart: (fieldKey, value) => string,
-        useLsid?: boolean,
-        singleFilterValue?: any
-    ) {
-        const keyCol = useLsid ? '/LSID' : '/RowId';
-        if (sampleColumns.size == 1) {
+    createSampleFilter(sampleColumns: List<string>, value, singleFilter: Filter.IFilterType) {
+        const keyCol = '/RowId';
+        if (sampleColumns.size === 1) {
             // generate simple equals filter
             const sampleColumn = sampleColumns.get(0);
-            return Filter.create(sampleColumn + keyCol, singleFilterValue ? singleFilterValue : value, singleFilter);
-        } else {
-            // generate an OR filter to include all sample columns
+            return Filter.create(sampleColumn + keyCol, value, singleFilter);
+        } else if (sampleColumns.size > 1) {
```

Comment:
Does the caller handle the case of an undefined return value from this method?

Reply:
sort of. It has a check before its usage to make sure `sampleColumns` is not empty. I'll make that a little cleaner.